### PR TITLE
3112 support generic multi framework route path convention and automatic conversion utility

### LINKED ISSUE #3112: Support generic multi-framework route path convention and automatic conversion utility
## Is your feature request related to a problem? Please describe.

Supporting multiple HTTP frameworks in Ts.ED (Express v4, Express v5, Koa, Fastify) is challenging due to differences in route wildcard and parameter syntax. For example, wildcards and named params require specific path patterns per framework, which leads to repetitive, non-portable code and increases the risk of subtle bugs when switching or supporting multiple backends.

## Describe the solution you'd like

- Define a generic, unified path convention for Ts.ED, such as `/foo/*` and `/foo/:bar*`, to describe wildcard and parameterized routes at the code/decorator level.
- Implement a utility to convert this generic path into the correct syntax for the target framework at runtime (Express v4, Express v5, Koa, Fastify).
- Preserve explicit regexp syntax (e.g., `/foo/:bar(.*)`, `/foo/(.*)`) and do not convert those paths.
- Provide unit tests for all cases.
- Bonus: Propose a decorator or integration to make this conversion transparent to Ts.ED users and document the convention in the official documentation.

### Example Implementation

```typescript
export type Framework = "express-v4" | "express-v5" | "koa" | "fastify";

export function convertRoutePath(path: string, framework: Framework): string {
  // If the path includes explicit regexp, do not convert
  if (/(\/:?[a-zA-Z0-9_]*\(\.\*\)|\/(\.\*))/ .test(path)) {
    return path;
  }
  switch (framework) {
    case "koa":
      return path
        .replace(/:([a-zA-Z0-9_]+)\*/g, ":$1(.*)")
        .replace(/\/\*$/, "/(.*)");
    case "express-v5":
      if (/\/\*$/.test(path)) {
        return path.replace(/\/\*$/, "/*splat");
      }
      return path;
    case "express-v4":
      return path;
    case "fastify":
      return path;
    default:
      throw new Error("Unknown framework: " + framework);
  }
}
```

### Unit Test Example

```typescript
describe("convertRoutePath", () => {
  it("should preserve explicit regexp syntax", () => {
    expect(convertRoutePath("/foo/:bar(.*)", "koa")).toBe("/foo/:bar(.*)");
    expect(convertRoutePath("/foo/(.*)", "express-v5")).toBe("/foo/(.*)");
  });
  it("should convert simple wildcards and named wildcards", () => {
    expect(convertRoutePath("/foo/*", "koa")).toBe("/foo/(.*)");
    expect(convertRoutePath("/foo/:bar*", "koa")).toBe("/foo/:bar(.*)");
    expect(convertRoutePath("/foo/*", "express-v5")).toBe("/foo/*splat");
    expect(convertRoutePath("/foo/:bar*", "express-v5")).toBe("/foo/:bar*");
    expect(convertRoutePath("/foo/*", "express-v4")).toBe("/foo/*");
    expect(convertRoutePath("/foo/:bar*", "express-v4")).toBe("/foo/:bar*");
    expect(convertRoutePath("/foo/*", "fastify")).toBe("/foo/*");
    expect(convertRoutePath("/foo/:bar*", "fastify")).toBe("/foo/:bar*");
  });
});
```

## Acceptance criteria

- [ ] A generic path convention is defined for Ts.ED route decorators.
- [ ] An automatic conversion utility adapts generic paths to Express v4, Express v5, Koa, and Fastify syntax.
- [ ] The utility preserves explicit regexp paths and does not alter them.
- [ ] Comprehensive unit tests cover all conventions and edge cases.
- [ ] (Bonus) Decorator/integration for transparent conversion in Ts.ED.
- [ ] (Bonus) Documentation update for the new convention and usage.

### PR DESCRIPTION
**Pull request by Romakita**
## Information

| Type                  | Breaking change |
| --------------------- | --------------- |
| Feature | No          |

---

## Todos

- [ ] Tests
- [ ] Coverage
- [ ] Example
- [ ] Documentation

## Summary by CodeRabbit

* **New Features**
  * Added comprehensive documentation on routing path conventions, including a new "Routing" guide and sidebar link.
  * Enhanced support for various route path patterns and wildcards across Express, Koa, and Fastify platforms.
  * Improved static asset route handling to support regular expressions.
  * Added new test endpoints demonstrating wildcard and optional path parameter usage.

* **Bug Fixes**
  * Ensured consistent extraction and mapping of wildcard parameters in route handlers for all supported platforms.

* **Documentation**
  * Expanded and clarified platform adapter and controller documentation, with new examples and formatting improvements.

* **Tests**
  * Introduced extensive test coverage for path conversion utilities and new route parameter patterns.

* **Style**
  * Minor formatting and code cleanup in documentation and codebase.

### CODE REVIEW

File: packages/platform/platform-express/src/utils/convertPath.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        return options;
+      }
+
+      if (segment.startsWith("{")) {
+        // Handle v5 style parameters like /{param}
+        const paramName = segment.substring(1, segment.length - 1);
+        options.path.push(`${paramName}?`);
+
+        return options;
+      }
```
[the remark this answers is not in the document]
Comment (Romakita) [reply packages/platform/platform-express/src/utils/convertPath.ts:46]: non le test couvre bien la syntaxe souhaité:
```
/{:test} => /:test?
```
